pint/facets/numpy/numpy_func.py

```diff
@@ -284,6 +284,18 @@ def implement_func(func_type, func_str, input_units=None, output_unit=None):
 
     @implements(func_str, func_type)
     def implementation(*args, **kwargs):
+        if func_str in ["multiply", "true_divide", "divide", "floor_divide"] and any(
+            [
+                not _is_quantity(arg) and _is_sequence_with_quantity_elements(arg)
+                for arg in args
+            ]
+        ):
+            # the sequence may contain different units, so fall back to element-wise
+            return np.array(
+                [func(args[0][i], args[1][i]) for i in range(len(args[0]))],
+                dtype=object,
+            )
```

Comment:
I wonder if it would be better to use `zip` instead? We're loading everything into memory, anyways:
```suggestion
            return np.array([func(*func_args) for func_args in zip(*args)], dtype=object)
```